### DISCUSSION TITLE
Fixing vague error messages cause new error

### DISCUSSION THREAD

**Discussion by daniel-ls-howard** [nextgenhealthcare/connect#6032] [Q&A]
Hi All,

A while ago I was trying to solve a problem where the logger wasn't specific enough when logging errors:

I found this post:

https://forums.mirthproject.io/forum/mirth-connect/support/14933-vague-error-message

That recommended setting the logger name based on the channel/destination names.

This worked fine under Java 11/Mirth 3.12 but under Java 17/Mirth 4.4.2 I get a script error:

```
297: 	logger = org.apache.log4j.Logger.getLogger(name);
298: 	logger.setLevel(level);
299: }
LINE NUMBER:	298
DETAILS:	Can't find method org.apache.log4j.Category.setLevel(org.apache.logging.log4j.Level).
	at dd39e227-5099-47d7-b93c-c1fbc9e90d2d_JavaScript_Filter_Transformer_3:298 (initLogger)



```

Any ideas?

TIA

**Comment by ab-mg-23** [comment]
This is due to change from log4j to log4j2. The following should let you set the logging level for a logger at runtime:
`org.apache.logging.log4j.core.config.Configurator.setLevel(name,level);`

**Comment by odoodo** [comment]
An alternative, holistic approach can be found [here](https://github.com/odoodo/Mirth-MetaAppender)
